std/http/server.ts

```diff
@@ -182,14 +219,107 @@ export class ServerRequest {
         }
         // TODO: handle other transfer-encoding types
       }
-      // Otherwise...
-      yield new Uint8Array(0);
+      // Otherwise... Do nothing
     }
   }
 
-  // Read the body of the request into a single Uint8Array
+  /**
+   * Read the request body as a stream.
+   * Currently, it can only handle basic body with content length,
+   * or `Transfer-Encoding: chunked`.
+   * @param bufferToReuse If provided, buffer to reuse for new data.
+   *   (actually returned buffer is a subarray of given buffer).
+   *   Returned buffer will be overwritten on next().
+   * @param bufferToFill If provided, buffer to fully filled.
+   *   Requires Content-Length header valid and buffer larger than it.
+   */
+  public async *bodyStream(
```

Comment:
@kevinkassimo could you explain those arguments? I'm not sure I understand why we need both of them

Reply:
They are mutually exclusive:

`bufferToReuse` is the buffer that would be overwritten every single time between yields. It is similar to the `.read` case except for that the buffer to write to every single time is fixed (maybe we can drop this if implementing Reader interface is good enough). Under this option, yielded buffer each time is a subarray always from start of `bufferToReuse` with length equal to bytes read at each step.

`bufferToFill` is only usable when the body has a clear content length, such that the user simply preallocate a buffer (or done through `.body()`) of large enough size to fit the complete body. Every slot of the buffer is written at most once. Under this option, yielded buffer each time is a subarray of `bufferToFill` that always starts from one byte after the previously yielded buffer.